### PR TITLE
Switching to NSX-T Simplified API

### PR DESCRIPTION
**Pull request by MartinWeindel**
**What this PR does / why we need it**:
VMware NSX-T is in the process to move from the old "Advanced API" to an intent-based "Simplified API". As there is no migration path from the old to the new API, we need to switch now to the new one to avoid the need to support both APIs later. 
The support for DHCP server is not complete in the current NSX-T 2.5. Therefore we still need to use the old API for this part. Using both APIs would be cumbersome with Terraform providers. The infrastructure controller does not use Terraform anymore, but creates the network infrastructure directly via API. The state is stored in the infrastructure status.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Please squash the commits into one.

**Release note**:

```noteworthy operator
The network infrastructure is created with the new "Simplified API" of NSX-T. 
```

```action operator
Shoots created with old versions of this provider-vsphere cannot be handled because of incompatible changes in the NSX-T object model and must be deleted before upgrading to this version
```

### CODE REVIEW

File: pkg/apis/vsphere/types_infrastructure.go [hunk 1]
```diff
@@ -57,9 +57,9 @@ type ZoneConfig struct {
 type InfrastructureStatus struct {
 	metav1.TypeMeta
 
-	Network         string
-	LogicalSwitchId string
-	LogicalRouterId string
+	SegmentName      string
```
[the remark this answers is not in the document]
Comment (MartinWeindel) [reply pkg/apis/vsphere/types_infrastructure.go]: yes segment name is required. it is used as network in vSphere. And yes, it is unique.

File: pkg/apis/vsphere/v1alpha1/types_cloudprofile.go [hunk 2]
```diff
@@ -176,4 +177,10 @@ type LoadBalancerClass struct {
 	Name string `json:"name"`
 	// IPPoolName is the name of the NSX-T IP pool.
 	IPPoolName string `json:"ipPoolName"`
+	// TCPAppProfileName is the profile name of the load balaner profile for TCP
+	// +optional
+	TCPAppProfileName *string `json:"tcpAppProfileName,omitempty"`
+	// UDPAppProfileName is the profile name of the load balaner profile for UDP
+	// +optional
+	UDPAppProfileName *string `json:"udpAppProfileName,omitempty"`
```
[the remark this answers is not in the document]
Comment (MartinWeindel) [reply pkg/apis/vsphere/v1alpha1/types_cloudprofile.go]: According to Rafael, this is the convention to be used for types.

[the remark this answers is not in the document]
Comment (rfranzke) [reply pkg/apis/vsphere/v1alpha1/types_cloudprofile.go]: Pointer types help if you want to differentiate whether a user specified the key in the manifest or not. If you use `string` here then omitting the key results in the same as if the user would have specified `udpAppProfileName: ""`.

File: pkg/controller/controlplane/valuesprovider.go [hunk 3]
```diff
@@ -372,26 +364,46 @@ func (vp *valuesProvider) getConfigChartValues(
...
+			//			}
+			if constraintClass != nil && constraintClass.IPPoolName != "" {
+				lbClass["ipPoolName"] = constraintClass.IPPoolName
 			}
-			return nil, fmt.Errorf("load balancer class %q not found in cloud profile", cpClass.Name)
 		} else {
 			lbClass["ipPoolName"] = *cpClass.IPPoolName
-			loadBalancersClasses = append(loadBalancersClasses, lbClass)
 		}
+		if cpClass.TCPAppProfileName == nil || *cpClass.TCPAppProfileName == "" {
```
[the remark this answers is not in the document]
Comment (MartinWeindel) [reply pkg/controller/controlplane/valuesprovider.go]: done

File: pkg/controller/infrastructure/actuator_delete.go [hunk 4]
```diff
@@ -16,26 +16,31 @@ package infrastructure
...
+	if prepared.state == nil {
+		// no state => nothing to do
+		return nil
+	}
+
+	state := prepared.state
+	err = prepared.ensurer.EnsureInfrastructureDeleted(state)
 	if err != nil {
-		return fmt.Errorf("could not create the Terraformer: %+v", err)
+		err2 := a.saveStateToConfigMap(ctx, infra.Namespace, state)
```
[the remark this answers is not in the document]
Comment (MartinWeindel) [reply pkg/controller/infrastructure/actuator_delete.go]: done

File: pkg/internal/infrastructure/tasks.go [hunk 5]
```diff
@@ -0,0 +1,733 @@
...
+func creatingErr(err error) (string, error) {
+	return "", errors.Wrapf(nicerVAPIError(err), "creating")
+}
+
+func toReference(s *string) *Reference {
+	if s == nil {
+		return nil
+	}
+	return &Reference{ID: *s}
+}
```
[the remark this answers is not in the document]
Comment (MartinWeindel) [reply pkg/internal/infrastructure/tasks.go]: done

**Pull request merged**